crates/eth-rpc/src/servers/alchemy_rpc.rs

```diff
@@ -22,8 +30,117 @@ impl<P: Provider + Send + Sync> AlchemyRpc<P> {
 
 #[async_trait]
 impl<P: Provider + Send + Sync + 'static> AlchemyApiServer for AlchemyRpc<P> {
+    async fn token_allowance(
+        &self,
+        contract_address: Address,
+        account_address: Address,
+        spender_address: Address,
+    ) -> Result<TokenAllowance> {
+        let entry_point: Felt252Wrapper = keccak256("allowance(account, spender)")
+            .try_into()
+            .map_err(|e: ConversionError<()>| EthApiError::<P::Error>::ConversionError(e.to_string()))?;
+        let entry_point: FieldElement = entry_point.into();
+
+        let account_addr: Felt252Wrapper = account_address.into();
+        let account_addr: FieldElement = account_addr.into();
+
+        let spender_addr: Felt252Wrapper = spender_address.into();
+        let spender_addr: FieldElement = spender_addr.into();
+
+        let calldata = vec![entry_point, account_addr, spender_addr];
+
+        let handle = self
+            .kakarot_client
+            .call(contract_address, Bytes::from(vec_felt_to_bytes(calldata).0), BlockId::from(BlockNumberOrTag::Latest))
+            .await;
+
+        let token_allowance = match handle {
+            Ok(call) => {
+                let allowance = U256::try_from_be_slice(call.as_ref())
+                    .ok_or(ConversionError::<()>::Other("error converting from Bytes to U256".into()))
+                    .unwrap();
+                TokenAllowance { result: Some(allowance), error: None }
+            }
+            Err(e) => TokenAllowance { result: None, error: Some(format!("kakarot_getTokenAllowance Error: {e}")) },
+        };
+
+        Ok(token_allowance)
+    }
+
     async fn token_balances(&self, address: Address, contract_addresses: Vec<Address>) -> Result<TokenBalances> {
         let token_balances = self.kakarot_client.token_balances(address, contract_addresses).await?;
         Ok(token_balances)
     }
+
+    async fn token_metadata(&self, contract_address: Address) -> Result<TokenMetadata> {
+        let decimals_entry_point: Felt252Wrapper = keccak256("decimals()")
+            .try_into()
+            .map_err(|e: ConversionError<()>| EthApiError::<P::Error>::ConversionError(e.to_string()))?;
```

Comment:
In case it doesn't work, we should probably open an issue to implement error conversion for these two error types, in order to minimize the use of map_err in our codebase.